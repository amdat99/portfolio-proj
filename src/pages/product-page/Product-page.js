import React,{Suspense} from 'react';

import ShopHeader from '../../components/shop-header/Shop-header'
import { addItem} from '../../redux/cart/cart.actions'


import { createStructuredSelector } from 'reselect';
import {connect} from 'react-redux'

import { selectProduct, isProductDataLoaded } from '../../redux/shop/shop.selectors'

const ProductItem = React.lazy(() => import('../../components/product-item/Product-item'))

function ProductPage({isLoaded,product, incrementItem}) {

    return (
        <div className="product-page-container">
        
        <ShopHeader />
        <Suspense fallback ={<div className="loader"></div>}>
        {  isLoaded
        ?product.map(item =>
         <ProductItem item={item} key={item.name} incrementItem={incrementItem}/> )
        
        : <h1>LOADING...</h1>
        } 
        
        </Suspense>
        </div>
    );
}

const mapStateToProps = (createStructuredSelector)({
    product: selectProduct,
    isLoaded: isProductDataLoaded
})

const mapDispatchToProps = (dispatch) => ({
    incrementItem: item => dispatch(addItem(item)),
})

 export default connect(mapStateToProps,mapDispatchToProps)(ProductPage);