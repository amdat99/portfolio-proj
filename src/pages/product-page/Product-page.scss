#map{
    position: relative;
    top:50px

}