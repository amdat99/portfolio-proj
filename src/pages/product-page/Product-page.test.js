import React from 'react';

import { shallow } from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import toJson from 'enzyme-to-json';

import {ProductPage} from './Product-page'

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

      
  
        let wrapper
   

        
  beforeEach(() => {

   const mockProps = {
            produt: {
              picture: 'www.picture.com',
              price: 10,
              name: 'ball',
              quantity: 5,
              profileId: 123
            },
          
          }; 
          
           wrapper = toJson(shallow(<ProductPage  {...mockProps} />))
           
        })
        



it('expect to render cartItem component', () => {
    expect(wrapper).toMatchSnapshot();
})

