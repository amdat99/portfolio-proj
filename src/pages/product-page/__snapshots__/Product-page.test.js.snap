// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render cartItem component 1`] = `
<div
  className="product-page-container"
>
  <Connect(ShopHeader) />
  <Suspense
    fallback={
      <div
        className="loader"
      />
    }
  >
    <h1>
      LOADING...
    </h1>
  </Suspense>
</div>
`;
